### PR TITLE
[media] Send specified error message for player capability changed errors

### PR DESCRIPTION
**Pull request by osagie98**
Also introduces `kSbPlayerErrorMessageCapabilityChanged` to replace the outgoing `kSbPlayerErrorCapabilityChanged`. Platforms can now report capability changed errors using the `kSbPlayerErrorDecode` code instead, while setting the error message to`kSbPlayerErrorMessageCapabilityChanged`.

b/321844872

### CODE REVIEW

File: cobalt/media/base/sbplayer_pipeline.cc [hunk 1]
```diff
@@ -1233,9 +1233,11 @@ void SbPlayerPipeline::OnPlayerError(SbPlayerError error,
     case kSbPlayerErrorDecode:
       CallErrorCB(::media::PIPELINE_ERROR_DECODE, message);
       break;
+#if SB_API_VERSION < 16
```
Comment (jasonzhangxx) [inline cobalt/media/base/sbplayer_pipeline.cc:1236]: How about removing PLAYBACK_CAPABILITY_CHANGED but still keep kSbPlayerErrorCapabilityChanged? We can report PIPELINE_ERROR_DECODE with a specified error message here for kSbPlayerErrorCapabilityChanged. So, under starboard, we can still use kSbPlayerErrorCapabilityChanged.

It's not a recommendation, more like a brainstorm of how we handle the difference between cobalt and chrome. Also @xiaomings.

Comment (osagie98) [reply cobalt/media/base/sbplayer_pipeline.cc:1236]: I think it's a good suggestion, so we can get closer to Chrome at the Cobalt level without making a Starboard change.

File: cobalt/media/base/sbplayer_pipeline.cc [hunk 2]
```diff
@@ -1234,7 +1234,16 @@ void SbPlayerPipeline::OnPlayerError(SbPlayerError error,
       CallErrorCB(::media::PIPELINE_ERROR_DECODE, message);
       break;
     case kSbPlayerErrorCapabilityChanged:
+#if SB_API_VERSION < 16
       CallErrorCB(::media::PLAYBACK_CAPABILITY_CHANGED, message);
+#else
+      CallErrorCB(::media::PIPELINE_ERROR_DECODE,
```
Comment (jasonzhangxx) [inline cobalt/media/base/sbplayer_pipeline.cc:1240]: nit: it's good to add a comment here to explain that we're removing non standard capability changed error.

And also let's notify h5 player team that we'll change capability changed error as they may need to refine their handling.

**Review by jasonzhangxx — APPROVED** [review]

File: cobalt/media/base/sbplayer_pipeline.cc (hunk 2 above)
Comment (xiaomings) [reply cobalt/media/base/sbplayer_pipeline.cc:1240]: We should probably ask the h5 player team to add the handling of the new error message now, in that case we may be able to remove all the `#if SB_API_VERSION < 16` guard in this PR.

Comment (osagie98) [reply cobalt/media/base/sbplayer_pipeline.cc:1242]: The H5 PR is in, so I've removed all the guards.

**Comment by xiaomings** [comment]
Just want to double check if we should remove `kSbPlayerErrorCapabilityChanged` from player.h for C25 (guarded by SB version)?

**Comment by osagie98** [comment]
> Just want to double check if we should remove `kSbPlayerErrorCapabilityChanged` from player.h for C25 (guarded by SB version)?

I left `kSbPlayerErrorCapabilityChanged` so Starboard can continue to send these errors, while Cobalt will handle reporting it as a decode error with message "MEDIA_ERR_CAPABILITY_CHANGED".

We can go ahead and remove it from Starboard from C25 on if we want to make it cleaner, it'll just require an interface change (as you've mentioned).

**Comment by xiaomings** [comment]
Sorry that Jason reminded me the original discussion and let's keep it as is.

And shall we remove MEDIA_ERR_CAPABILITY_CHANGED from media_error.idl?

**Comment by osagie98** [comment]
Yes, [pull request here](https://github.com/youtube/cobalt/pull/2608)

**Pull request merged**